### PR TITLE
version: make regexes multiline

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

This PR converts the regular expressions in `version.rb` to be multiline, which I admit, might seem ugly at first. However, I think that this change is desirable as it makes it more obvious what each individual part is doing. (In fact, I found a potential bug while working on this PR and added it as a TODO comment!) Granted, I may have gone overboard in some cases, so please tell me when you feel that is the case.

I tried not to make any changes to the regular expressions themselves other than adding newlines, indentation, and line comments. As it would be too easy to make a small mistake, I think any changes other than whitespace or comments should be made in follow-up PRs.

### CODE REVIEW

File: Library/Homebrew/version.rb [hunk 1]
```diff
@@ -363,7 +363,7 @@ def self.parse(spec, detected_from_url: false)
   end
 
   sig { params(spec: T.any(String, Pathname), detected_from_url: T::Boolean).returns(T.nilable(String)) }
-  def self._parse(spec, detected_from_url:)
+  def self._parse(spec, detected_from_url:) # rubocop:disable Metrics/MethodLength
```
Comment (MikeMcQuaid) [inline Library/Homebrew/version.rb:366]: Can we either change the rule or refactor the method to use e.g. constants?

**Review by MikeMcQuaid — COMMENTED** [review]
> This PR converts the regular expressions in version.rb to be multiline, which I admit, might seem ugly at first. However, I think that this change is desirable as it makes it more obvious what each individual part is doing. (In fact, I found a potential bug while working on this PR and added it as a TODO comment!) Granted, I may have gone overboard in some cases, so please tell me when you feel that is the case.

Unfortunately: this does feel a bit overboard to me. To me, multiline regex are useful primarily when you are commenting multiple lines to explain what's going on. I'm not saying you should do that here but if you're willing to: that's the point at which I'd agree the readability improvements would make this change worthwhile.

**Comment by SeekingMeaning** [comment]
To propose an alternative: what about adding constants for sub-regexes? For example:

- `(?:\d+\.)*\d+` is present in 7 of the 24 regexes, and 10 others each has some other similar variant
- `(?i:bin|dist|stable|src|sources?|final|full)` is present in 2 regexes
- `(?i:alpha|beta|pre|rc)` is present in 2 other regexes

**Comment by SeekingMeaning** [comment]
Actually, I think I'll do that and postpone this for another time

**Pull request closed without merging**